mediacapture-streams/MediaDevices-enumerateDevices-returned-objects.https.html

```diff
@@ -28,12 +30,31 @@
             assert_not_equals(device1, device2);
             assert_equals(device1.deviceId, device2.deviceId, "deviceId");
             assert_equals(device1.kind, device2.kind, "kind");
+            if (!callGetUserMedia) {
+              /* For camera and microphone devices,
+               if the browsing context did not capture (i.e. getUserMedia() was not called or never resolved successfully),
+               the MediaDeviceInfo object will contain a valid value for kind
+               but empty strings for deviceId, label, and groupId. */
+              assert_equals(device1.deviceId, "", "deviceId is empty before capture");
+              assert_equals(device1.groupId, "", "groupId is empty before capture");
+              assert_equals(device1.label, "", "label is empty before capture");
+            }
+        }
+        /* Additionally, at most one device of each kind
+           will be listed in enumerateDevices() result. */
+        // FIXME: ensure browsers are tested as if they had multiple devices of at least one kind -
+        // this probably needs https://w3c.github.io/mediacapture-automation/ support
+        if (!callGetUserMedia) {
+            const deviceKinds = deviceList1.map(d => d.kind);
+            for (let kind of deviceKinds) {
+              assert_equals(deviceKinds.filter(x => x===kind).length, 1, "At most 1 " + kind + " prior to capture");
+            }
         }
     }, testName);
 }
 
-doTest(false, "enumerateDevices returns new objects in case device-info permission is not granted");
-doTest(true, "enumerateDevices returns new objects in case device-info permission is granted");
+doTest(false, "enumerateDevices returns expected mostly empty objects in case device-info permission is not granted");
+doTest(true, "enumerateDevices returns expected objects in case device-info permission is granted");
```

Comment:
I think this would be easier to read if it were flattened into two independent tests.

Reply:
I don't disagree, but since the test started that way, I don't feel the urgent need to flatten it :)